### PR TITLE
Respect addSpaceOnEdges when initializing axes

### LINKED ISSUE #687: addSpaceOnEdges is still not respected
#645 seems to be an issue still. 

- [ ] Feature request
- [X] Design defect
- [ ] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version:
0.32.12
### Issue description
addSpaceOnEdges is not respected when a timescale 
#### Steps to produce the issue
Create any timescale graph with a larger timeframe (>18h, rough guess) or possibly a larger number of data points in that span? Codebox link provides sample data to repro.

#### Current behavior
Space is padded on the graph.

#### Expected behavior
No space, as configured.

### Screenshot or recording
https://codesandbox.io/s/exciting-margulis-6c1ep?file=/App.svelte:15253-15304

### PR DESCRIPTION
**Pull request by theiliad**
Closes #687

### CODE REVIEW

File: packages/core/src/components/axes/axis.ts [hunk 1]
```diff
@@ -186,20 +193,24 @@ export class Axis extends Component {
...
 					// If there are more than 2 labels to show
-					if (
-						Tools.getProperty(
-							options,
-							"timeScale",
-							"addSpaceOnEdges"
-						) &&
-						tickValues.length > 2
-					) {
+					if (addSpaceOnEdges && tickValues.length > 2) {
```
Comment (theiliad) [inline packages/core/src/components/axes/axis.ts:213]: prettier seems to have added a lot of changes, this is what I've changed logic wise in the code

Comment (theiliad) [reply packages/core/src/components/axes/axis.ts:213]: @natashadecoste @Donisius @cal-smith

File: packages/core/src/components/axes/axis.ts [hunk 2]
```diff
@@ -186,20 +193,24 @@ export class Axis extends Component {
...
+					);
+
+					let tickValues;
+					if (addSpaceOnEdges) {
+						tickValues = scale
+							.nice(numberOfTicks)
+							.ticks(numberOfTicks);
+					} else {
+						tickValues = scale.ticks(numberOfTicks);
+					}
```
Comment (Donisius) [inline packages/core/src/components/axes/axis.ts:209]: Both the if and else blocks have `scale.ticks(numberOfTicks);`. Is it somehow possible to bring that portion outside so we can get rid of the else block?

Comment (theiliad) [reply packages/core/src/components/axes/axis.ts]: 👍

**Comment by dstaley41** [comment]
Awesome, thanks for the quick fix!

**Pull request merged**